src/main/java/teammates/ui/controller/InstructorCourseStudentDetailsEditSaveAction.java

```diff
@@ -53,8 +53,19 @@ public ActionResult execute() throws EntityDoesNotExistException {
         student.comments = Sanitizer.sanitizeTextField(student.comments);
         
         try {
-            student.updateWithExistingRecord(logic.getStudentForEmail(courseId, studentEmail));
-            logic.validateSections(Arrays.asList(student), courseId);
+            StudentAttributes originalStudentAttribute = logic.getStudentForEmail(courseId, studentEmail);
+            student.updateWithExistingRecord(originalStudentAttribute);
+            
+            boolean hasSectionChanged = student.hasSectionChanged(originalStudentAttribute);
+            boolean hasTeamChanged = student.hasSectionChanged(originalStudentAttribute);
+            if(hasSectionChanged && hasTeamChanged){
+                logic.validateSectionsAndTeams(Arrays.asList(student), courseId);
+            } else if(hasSectionChanged){
```

Comment:
If the section has changed, we still need to validate the teams, as we do not allow same team names across different sections. For example, assuming Student A is in Section 1 Team 1, when changing Student A's section from Section 1 to Section 2, we need to ensure that there does not exist a Team 1 in another Section.